src/compiler/parser.ts

```diff
@@ -2855,7 +2862,7 @@ namespace ts {
             return finishNode(factory.createQualifiedName(entity, name), entity.pos);
         }
 
-        function parseRightSideOfDot(allowIdentifierNames: boolean, allowPrivateIdentifiers: boolean): Identifier | PrivateIdentifier {
+        function parseRightSideOfDot(allowIdentifierNames: boolean, allowPrivateIdentifiers: boolean, allowUnicodeEscapeSequenceInIdentifierName = true): Identifier | PrivateIdentifier {

```

Comment:
Optional parameters are usually a bad idea in the parser, because it makes it easy to forget when a new call to this function may need to pass `true`. I would recommend making this a required parameter like the others.